docs/notebooks/demo/navigator/navigator-data-designer-sdk-multi-turn-conversation.ipynb

```diff
@@ -0,0 +1,429 @@
+    ")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Adding Generated Data Columns\n",
+    "We now define the columns that the model will generate. These prompts instruct the LLM to produce the actual conversation: a user query (user_message), the assistant’s response (assistant_message), a follow-up user query (user_message_2), and another assistant response (assistant_message_2). Finally, we generate a toxicity_label to assess user toxicity over the entire conversation.\n",
+    "\n",
+    "You can easily modify or refine these prompt templates to adjust the style, complexity, or constraints of the generated data. Maintaining continuity and consistency across turns ensures the dialogues are realistic and useful for fine-tuning.\n",
+    "\n",
+    "#### Turn 1: User Message and Assistant Response\n",
+    "\n",
+    "In the first turn, we define the user_message column to simulate the user's initial query and then the assistant_message column for the assistant's reply. Ensuring that the assistant message does not always start the same way helps produce more natural variations."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# User initial message\n",
+    "data_designer.add_generated_data_column(\n",
+    "    name=\"user_message\",\n",
+    "    generation_prompt=(\n",
+    "        \"The user is seeking help or information in the {domain} domain, specifically about the topic of {topic}, \"\n",
+    "        \"at a {complexity} complexity level.\\n\\n\"\n",
+    "        \"The user_message should:\\n\"\n",
+    "        \"- Sound natural and realistic.\\n\"\n",
+    "        \"- Avoid disallowed content.\\n\"\n",
+    "        \"- Reflect the specified domain, topic, and complexity level.\\n\"\n",
+    "        \"- Do not include headers, explanations, or assessments.\\n\"\n",
+    "        \"- Do not include formatting like '## ...' or similar.\\n\"\n",
+    "    )\n",
+    ")\n",
+    "\n",
+    "# Assistant responds\n",
+    "data_designer.add_generated_data_column(\n",
+    "    name=\"assistant_message\",\n",
+    "    generation_prompt=(\n",
+    "        \"As a helpful assistant, write a response to the user's query below:\\n\"\n",
+    "        \"Query: {user_message}\\n\\n\"\n",
+    "        \"Instructions:\\n\"\n",
+    "        \"- Provide a clear, accurate, and contextually relevant response.\\n\"\n",
+    "        \"- Be correct, non-toxic, and helpful.\\n\"\n",
+    "        \"- Avoid disallowed content. If the request is disallowed, provide a safe refusal.\\n\"\n",
+    "        \"- To encourage variety, do not always start your response with the same phrase.\\n\"\n",
+    "        \"- Only provide the message, do not add headers, explanations, or assessments.\\n\"\n",
+    "        \"- Do not include formatting like '## ...' or similar.\\n\"\n",
+    "    )\n",
+    ")"
+   ]
+  },
```

Comment:
It's technically possible to use `output_type=dict` to do these in one shot, but our current parsing solution isn't robust. Big improvements coming soon!